haystack/document_stores/opensearch.py

```diff
@@ -1333,7 +1337,12 @@ def _train_ivf_index(
             index = self.index
 
         # Check if IVF index is already trained by checking if embedding mapping contains a model_id field
-        if "model_id" in self.client.indices.get(index)[index]["mappings"]["properties"][self.embedding_field]:
+        if (
+            "model_id"
+            in self._client_indices_get(index=index, headers=headers)[index]["mappings"]["properties"][
+                self.embedding_field
+            ]
+        ):
```

Comment:
That `[index]["mappings"]["properties"][self.embedding_field]` seems quite dangerous, shall we wrap that in a `try`?

Reply:
I think there's no need to wrap it in a `try`, as the method should only be called after for already existing indexes.